src/devices/bus/centronics/epson_lx810l.cpp

```diff
@@ -399,7 +405,8 @@ uint8_t epson_lx810l_device::porta_r(offs_t offset)
 	LOG("%s: lx810l_PA_r(%02x): result %02x\n", machine().describe_context(), offset, result);
 
 	// Update the printhead display, only put this here since this routine gets called frequently
-	m_bitmap_printer->set_printhead_color( m_e05a30->ready_led() ? 0x55ff55 : 0x337733, 0x0);
+//  m_bitmap_printer->set_printhead_color( m_e05a30->ready_led() ? 0x55ff55 : 0x337733, 0x0);
+	m_bitmap_printer->set_led_state(0, !ioport("PAPEREND")->read());
```

Comment:
This should be changed to use a required_ioport device so we aren't doing string-based device lookup at runtime.  I use those all over the A2 drivers, look for m_kbspecial in apple2e.cpp for instance.

Reply:
ok, I changed every one I could find